doc: matter: Extended low power configuration with ICD information (#13038)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/getting_started/low_power_configuration.rst

* :ref:`Matter window covering sample <matter_window_covering_sample>`
* :ref:`Matter weather station application <matter_weather_station_app>`

.. _ug_matter_device_low_power_icd:

Enable Matter Intermittently Connected Devices support
******************************************************

> The Matter treats power-optimized devices as a part of the wider group, which are devices that are intermittently connected to the network or unreachable for periods of time.

Reviewer: Combining this with mine:
```suggestion
In Matter, power-optimized devices are part of the wider group of devices which includes devices that are intermittently connected to the network or otherwise unreachable for periods of time.
```

Active mode is a state, where the device is reachable and will answer in a timely manner for minimum period of time specified by the active mode duration (previously called active mode interval) parameter.
In the active mode, the device configures the networking interface in a fast polling, which means it is able to receive messages with the fastest frequency of fast polling interval.
The client of ICD has to respect the slow polling and fast polling intervals and should not try to send messages with the bigger frequency.

To configure the ICD communication intervals, use the following Kconfig options:

* :kconfig:option:`CONFIG_CHIP_ICD_IDLE_MODE_DURATION` to set idle mode duration value in seconds (for example, ``120``).
* :kconfig:option:`CONFIG_CHIP_ICD_ACTIVE_MODE_DURATION` to set active mode duration value in milliseconds (for example, ``300``).
* :kconfig:option:`CONFIG_CHIP_ICD_SLOW_POLL_INTERVAL` to set slow polling interval value in milliseconds (for example, ``5000``).
* :kconfig:option:`CONFIG_CHIP_ICD_FAST_POLLING_INTERVAL` to set fast polling interval value in milliseconds (for example, ``200``).

> An ICD switches modes from idle to active for various reasons that are:

Reviewer: Good point. If the list is not exhaustive, the line can end with ", such as:"

> An ICD switches modes from idle to active for various reasons that are:

Author: Actually I think these are the only reasons copied from Matter spec, because it treats "an application event" as everything else specific to the application. Nevertheless I think adding "such as" statement will not harm.

Once the ICD spend in the active mode period of time equal to the active mode duration and there are no further pending exchanges, it is allowed to switch to the idle mode to save the power.
In some scenarios it may be beneficial for the device to proactively spend more time in the active mode, instead of going to idle state immediately.
The reason to use this functionality is to make the device wait for potentially delayed incoming traffic and avoid energy-wasting retransmissions.

The period of time that ICD additionally stays in the active mode after finishing network activity is called active mode threshold and it can be configured using :kconfig:option:`CONFIG_CHIP_ICD_ACTIVE_MODE_THRESHOLD` Kconfig option.

The higher the threshold value, the greater the communication reliability, but it also leads to higher power consumption.
It can be useful mainly for an ICD using long slow polling interval, typically bigger than a few seconds, so getting another chance to receive a message will happen after a long time.
For the devices using short slow polling interval, using this functionality does not seem to be reasonable and may lead to unnecessary looses of energy.

> Short idle time and long idle time devices

Reviewer: We should follow the capitalization that the Matter SDK uses, whatever that is.

> Short idle time and long idle time devices

Author: Ok, so I will use capital letters for all parameter names.

==========================================

> The Matter v1.2 introduced division of ICD on short idle time (SIT) and long idle time (LIT) devices.

Reviewer: Combining this with mine:
```suggestion
The Matter v1.2 specification divides ICD into short idle time (SIT) and long idle time (LIT) devices.
```

The criteria for this division is idle mode duration parameter, which should be equal or shorter than 15 seconds for SIT and longer than 15 seconds for LIT.

> The SIT device behavior is aligned with formerly called SED in Matter v.1.1.

Reviewer 2: ```suggestion
The SIT device behavior is aligned with former SED from Matter v.1.1.
```

* Processing an attribute change that has to be reported to a subscriber.
* Processing an application event.

Active Mode Threshold
=====================

An ICD is allowed to switch to the idle mode to save power once it has spent time in the active mode equal to the active mode duration and there are no further pending exchanges.
In some scenarios it can be beneficial for the device to spend more time in the active mode instead of going to the idle mode immediately.

The period of time that an ICD additionally stays in the active mode after finishing network activity is called the Active Mode Threshold and it can be configured using the :kconfig:option:`CONFIG_CHIP_ICD_ACTIVE_MODE_THRESHOLD` Kconfig option.
Use this functionality to make the device wait for potentially delayed incoming traffic and to avoid energy-wasting retransmissions.

The higher the threshold value, the higher the communication reliability, but the power consumption is also higher.
A high threshold is useful mainly for an ICD with a long Slow Polling Interval, typically bigger than a few seconds, where getting another chance to receive a message will happen after a long time.

> For devices with a short Slow Polling Interval, using this functionality does not seem to be reasonable and may lead to unnecessary looses power consumption.

Reviewer: ```suggestion
For devices with a short Slow Polling Interval, using this functionality can lead to unnecessary power consumption.
```
We don't need to a value judgement in the documentation.

Both types are variants of Minimal Thread Device (MTD) type.
For more information about MTD and SEDs, see :ref:`Thread device types <thread_ot_device_types>`.

> The Matter ICD leverages Thread SED behavior, however these two terms are not necessarily related to each other and should not be confused.

Author: I think the previous version described the relation more accurately. The ICD and SED are different concepts describing behavior on a different abstraction level. Nevertheless in case of Thread, usage of ICD in Matter sort of implies enabling SED on a Thread network layer.

> The Matter ICD leverages Thread SED behavior, however these two terms are not necessarily related to each other and should not be confused.

Reviewer: Would this work
```suggestion
In a Thread network, :ref:`Matter ICD <ug_matter_device_low_power_icd>` behave like Thread SED, but the two terms are not interchangeable.
```
